### ISSUE TITLE
Wrong geometries in France

### ISSUE THREAD

**Issue by Joxit** [whosonfirst-data/whosonfirst-data#1094]
Hi,
The geometries of quattroshapes are inaccurate in some places.
For example here we have [Villetaneuse](https://spelunker.whosonfirst.org/id/101751173) (in red) and [Montmagny](https://spelunker.whosonfirst.org/id/101750079/) (in blue) with borders coming from OpenStreetMap (black lines).

![montmagny-villetaneuse](https://user-images.githubusercontent.com/5153882/37880544-66855a7e-308a-11e8-904f-ca674322b98c.png)

We can see very well that there is a part of Montmagny which is on the territory of Villetaneuse.
Because of this, some homes are misclassified with Pelias when using pip-whosonfirst during imports.
Would it be possible to retrieve geometries from OSM to update Who's On First?
Or are they too accurate and will be too heavy for Who's On First?

**Comment by adelcasse** [comment]
I have even an even stranger issue with Giromagny https://www.openstreetmap.org/relation/114117 and Rougegoutte https://www.openstreetmap.org/relation/114233#map=14/47.7335/6.8610 where Giromagny locality in WOF gets "out of" Giromagny localadmin and includes a part of Rougegoutte.

It might be interesting to import geometries maintained by the french government : https://www.data.gouv.fr/fr/reuses/geozones/ ("frgov" source https://github.com/whosonfirst/whosonfirst-sources/blob/eefa009180d1c6c9a5b604c947ff66da97e06224/sources/frgov.json only imports data related to postcodes but there are more complete data available)

**Comment by adelcasse** [comment]
If there is interest (and some documentation on how to do that correctly ;) ), I could integrate the governmental data in WOF and send a PR ...

**Comment by stepps00** [comment]
Hey @Joxit and @adelcasse - thanks for the issue and info..

@Joxit - We are not able to ingest OSM data, as WOF requires a `CC0` or `CC-BY` equivalent open license. The government of France maintains an open data portal though, that may contain useful data that we can use to update our records and fix these boundary issues.

@adelcasse - thanks for the links! We have documentation on updating neighbourhood records in Who's On First ([here](https://github.com/whosonfirst/whosonfirst-cookbook/blob/master/issue_workflows/sf_neighbourhood_updates_pt_1.md)) - the same concept and rules apply. Unfortunately the documentation lost some screenshots and images, but I can update the documentation and/or help with any questions you may have if you submit a PR.

**Comment by adelcasse** [comment]
Is french government's "Licence ouverte / open licence" compatible with WOF ? https://www.etalab.gouv.fr/licence-ouverte-open-licence (it says it is "compatible with common open licences standards including UK's open governement licence, ODC-BY, CC-BY 2.0) If it is, I can work on merging this dataset https://www.data.gouv.fr/fr/datasets/contour-des-iris-insee-tout-en-un/ (others are in ODBL si not usable in WOF ?) by next week.

**Comment by stepps00** [comment]
If the license terms are in line with what we have catalogued from a previous import (CC-BY 2.0), [here](https://github.com/whosonfirst/whosonfirst-sources/blob/eefa009180d1c6c9a5b604c947ff66da97e06224/sources/frgov.json), then yes, we are able to import the data into WOF. If the license requires attribution or restricts certain use cases, then we can not import the data.

**Comment by adelcasse** [comment]
Great, it is the same licence.
I searched for other datasets and there is https://www.data.gouv.fr/fr/datasets/contours-geographiques-des-nouvelles-regions-metropole-reunion-guadeloupe/ for regions (I didn't check yet if regions have a correct shape un WOF or not but de will have the reference if needed) https://www.data.gouv.fr/fr/datasets/contours-geographiques-des-nouvelles-regions-metropole-reunion-guadeloupe/

**Comment by stepps00** [comment]
The thirteen region divisions in France are actually "macroregions" in Who's On First - [link](https://spelunker.whosonfirst.org/id/85633147/descendants/?placetype=macroregion&exclude=nullisland&is_current=1). We could update geometries at this level with the `frgov` geometries, but the more pressing issue is updating the `localadmin` records. If you'd like to take a shot at updating both placetypes, feel free to do so!

Note that the source geometries do have some erroneous nodes from a feature dissolve:
<img width="952" alt="screen shot 2018-03-27 at 5 33 46 pm" src="https://user-images.githubusercontent.com/18567700/38002053-0b858732-31e5-11e8-9eeb-f676b44832c9.png">

**Comment by adelcasse** [comment]
Yes, even for our use cases, localadmins is the priority as we need them in Pelias, just like @Joxit ... It was just to give the reference and If it's not too long I'll take a look while updating localadmins

**Comment by Joxit** [comment]
@stepps00  Seriously? We can't use OSM as a source for WOF? It's sad... They're still in your list of sources [here] (https://github.com/whosonfirst/whosonfirst-sources/tree/master/sources#openstreetmap).

@adelcasse Nice catch for iris dataset. You will have to merge the geometries to find the localities because it corresponds to the neighborhoods. If you need help, I can take care of it too.

**Comment by stepps00** [comment]
@Joxit - yes, we use OSM for property prefixes (concordances) due to the `ODbL` license.

**Comment by Joxit** [comment]
@stepps00  Oh, ok, I understand. Thanks for your precisions :).

@adelcasse I compared your city between OSM and Iris, it's not perfect but good enought :). 
white is Iris and purple is OSM.
![Giromagny](https://user-images.githubusercontent.com/5153882/38051550-f583d5a2-32ce-11e8-95ef-25f42e1ff1d4.png)

**Comment by nvkelso** [comment]
The OSM shapes might be from French open data via http://www.dgi.gouv.ci/site/.

References:

- https://wiki.openstreetmap.org/wiki/WikiProject_Cadastre_Fran%C3%A7ais/Conditions_d%27utilisation
- https://wiki.openstreetmap.org/wiki/WikiProject_France/Limites_administratives
- https://wiki.openstreetmap.org/wiki/Potential_Datasources/France#DGFiP.2FCadastre
- http://openstreetmap.fr/36680-communes
- https://www.openstreetmap.org/copyright

**Comment by adelcasse** [comment]
@nvjekso this is the ivory cost, not France ;) on your referenced links, they say they got the data from DGFIP with à special access

**Comment by adelcasse** [comment]
I think the best dataset in "licence Ouverte / Open Licence" should be this one in fact : https://www.data.gouv.fr/fr/datasets/admin-express/

It includes all administrative layers available

What do you think @Joxit ?


For reference just in case, some more datasets in "Licence ouverte / Open licence" that could help on cities shapes (often derived from the previous one or its predecessor GeoFLA and provided by regions / metropolitan areas) : 

- Métropole de Rennes : https://www.data.gouv.fr/fr/datasets/communes-france-metropole-limites/#_
- Métropole de Lille : https://www.data.gouv.fr/fr/datasets/limites-administratives-des-communes-de-la-mel/
- Ile de France : https://data.iledefrance.fr/explore/dataset/cdgt2013_communes/
- "Pays Ajaccien" (part of Corsica) : https://www.data.gouv.fr/fr/datasets/limites-communales-des-communes-du-pays-ajaccien/
- Métropole de Bordeaux : https://www.data.gouv.fr/fr/datasets/limite-des-communes-de-la-metropole-de-bordeaux/ / https://www.datalocale.fr/dataset/limite-des-communes-de-la-metropole-de-bordeaux
- Bourgogne France Compté : https://www.data.gouv.fr/fr/datasets/communes-de-bourgogne-franche-comte-au-1-janvier-2018/

Data where licence is not specified (but should be open licenced as derived from GeoFLA http://professionnels.ign.fr/geofla ) : 
- https://www.data.gouv.fr/fr/datasets/communes-geofla-en-normandie/#resource-569f0449-e386-4014-9669-93eb3d4bf438

**Comment by Joxit** [comment]
@adelcasse  Nice, that's perfect ! 
I checked some admins with [admin-express](http://professionnels.ign.fr/adminexpress), (Giromagny, villetaneuse, paris, Martinique, L'île de la Réunion) they are perfect ! :smile: 
- Your link tagged as Métropole de Rennes is the GeoJSON version of admin-express.
- Métropole de Lille seems more accurate than admin-express.
- Ile-de-France is too simplified, bad quality.
- Bordeau is Good, look like OSM data.

(I can't test the other links, I got 504  Time-out errors ^^')

**Comment by adelcasse** [comment]
@stepps00 I've made a merge between admin-express and WOF data for localities and localadmins anywhere I could. I now have 2 geojson files (1 for all localities shapes updates, 1 for all localadmins shapes updates). Is there a way to merge this automatically with existing whosonfirst-data geojson files ? I didn't see documentation on this.

https://cloud.scity.coop/index.php/s/AUI1Uqt4FduzmGC/download

And anyway, could you tell me if it looks right or if I did something wrong (first time I try to play with WOF data ...) ?

**Comment by stepps00** [comment]
Hey @adelcasse and @joxit - thanks for the links and updated GeoJSON files.

A few comments/questions, below:

1. I see two links to the "admin-express" data - one [here](https://www.data.gouv.fr/fr/datasets/admin-express/) (under an "open license") and one [here](http://professionnels.ign.fr/adminexpress) (under a paid/subscription license). The data in the file uploaded by @adelcasse looks like it is sourced from the data.gouv.fr "open license" dataset, is this true?

2. The updated localadmin and locality files from @adelcasse contain partial coverage in France; there are holes in both placetypes. Is this expected? (See below, blue for localadmin, yellow for locality):

<img width="932" alt="screen shot 2018-04-03 at 10 04 22 am" src="https://user-images.githubusercontent.com/18567700/38263979-75de406a-3726-11e8-8f93-1dc0239ef47a.png">

Ideally, we will update an entire placetype for France to make sure we don't have overlapping geometries.

3. The properties in the localadmin and locality files look good - we can craft a script to import the source properties, move existing geometries to alt geometries, and update the existing records with new geometries (and label positions).

Let me know if these coverage gaps are expected -- I expect they are not. If they are, we'll have to figure out what to do with non-updated records.

**Comment by adelcasse** [comment]
@stepps00 it was "expected". I've selected localadmins and localities I could match between WOF and "admin express". I ignored the others for now (the ones I couln't join by just fixing accents/lower-uppercases discrepencies between both datasets). So I removed from the frgov datasets the admins without wof:id. I can try to get further to "fill the gaps" (but it might be difficult due to the amount of admins concerned), or should we just delete the wof:id we couln't match with and recreate localadmins/localities with "new" admins from frgov ?

On the licences, I'll double check but I think it's "open licence" in both places (etalab and IGN). But we might have to source IGN instead of frgov (IGN is a governemental agency so not sure)

**Comment by adelcasse** [comment]
On this link http://professionnels.ign.fr/adminexpress it says : "La réutilisation de ADMIN EXPRESS est gratuite pour tous les usages, y compris commerciaux, selon les termes de la "licence ouverte" version 2.0." which means "Reuse of Admin Express is free for all usages, including commercial, under the terms of the "Open Licence / Licence ouverte" version 2".

It is just near the "open licence/licence ouverte" logo from Etalab. Si it's exactly the same licence in both sources

(Etalab = data.gouv)

**Comment by Joxit** [comment]
@adelcasse be careful because your data uses Lambert projection and not WSG84 and there are some bad matches. For example for the wof id [101874597 (Saint-Sulpice)](https://spelunker.whosonfirst.org/id/101874597/) you got 8 localities matches (INSEE_COM : 01387, 60598, 63399, 70474, 53254, 58269, 73281, 46294)
You can get the output with : 
```sh
grep '"id": 101874597'  new_localities2.geojson
```

@stepps00 for gaps, there have been many new cities since the last wof updates. That why @adelcasse can't found every localities. Here is a list for new cities : https://fr.wikipedia.org/wiki/Liste_de_communes_nouvelles.

**Comment by Joxit** [comment]
I created a project for shapefile reading with WOF mapping. It is located here: [Joxit/whosonfirst-utils](https://github.com/Joxit/whosonfirst-utils) with an example [admin-express.js](https://github.com/Joxit/whosonfirst-utils/blob/master/examples/).
I executed it on admin express and here is the result: [result wof-fr-admin-express.zip](http://media.r7.jawg.io/wof-utils/wof-fr-admin-express.zip)
You have 4 output files: `localities.geojson`,` localadmin.geojson`, `localityAndLocaladmin.geojson`,` newlocalities.geojson`.
The interesting properties are: (`wof:locality_id` and `wof:localadmin_id`)

That's how it works:
1. Loading WOF data (or using an external pip-service)
2. Play shapefile, feature by feature and reproject in WSG84
3. For each feature, take a point on feature
4. Make a wof-lookup on this point (theoretically we must get the correct wof localadmin/locality associated with this point)
5. Look at the wof results names of locality and localadmin
    - If shapefile feature name match wof locality AND localadmin, then add both id and put the shapefile feature in the file `localityAndLocaladmin.geojson`
    - If shapefile feature name matches only one, add the id and put the shapefile feature either in `localities.geojson` or` localadmin.geojson`
    - If shapefile feature name does not matches, it's probably a new city, add it in `newlocalities.geojson` for a manual check or with wikipedia.

With the 4 files there will be no more gaps, next step will be to identify cities that no longer exist and remove them from wof.

**Comment by stepps00** [comment]
First off @joxit and @adelcasse - thank you both for untangling the locality and localadmin data in France. I'm reviewing the most recent files and the geometries look much better than what is currently in WOF. I need some time to review each of the four geojson files, craft an import script and plan, and submit a PR.. but once that is complete, I'll update this issue.

@joxit - to make sure we're on the same page, here are my notes about these files:

- `localityandlocaladmin.geojson` - Source features that match the name of **both** a WOF locality and localadmin record.
- `locality.geojson` - Source features that match the name of a WOF locality record.
- `localadmin.geojson` - Source features that match the name of a WOF localadmin record.
- `newlocalities.geojson` - Source features that **do not** match either a WOF locality or localadmin record.

I have noticed, though, that some features in the `newlocalities.geojson` file are actually in WOF. An example is "Goudelancourt-lès-Berrieux" -- WOF has a [locality](https://spelunker.whosonfirst.org/id/1126055357/) and [localadmin](https://spelunker.whosonfirst.org/id/1125326895/) record for this place. Any idea how this feature made it to the `newlocalities.geojson` file?

I'm also going to compare the source files and WOF data to make sure we catch all matches.

**Comment by Joxit** [comment]
@stepps00  Your are welcome :smile: Ok thx

Yes, it's exactly that :smile:.

Oh, you are right.... That's because pip-service is returning this, the accent is missing
```json
{
    "localadmin": [
        {
            "bounding_box": "3.82871046804,49.4798743409,3.8858445711,49.5056131933", 
            "centroid": { "lat": 49.494677, "lon": 3.858589 }, 
            "id": 404423617, 
            "name": "Goudelancourt-Les-Berrieux"
        }
    ]
}
```
I will try to fix that

**Comment by adelcasse** [comment]
@Joxit I guess there is an issue with accents explaining the "Goudelancourt-lès-Berrieux". While WOF localadmins don't use accents in their names, WOF localities do. I had the same issue on my side.

I used these replaces on localadmins : you might probably adapt this in your script at comparison time :


sed -i "s/é/e/g" $1
sed -i "s/è/e/g" $1
sed -i "s/ê/e/g" $1
sed -i "s/â/a/g" $1
sed -i "s/-l/-L/g" $1
sed -i "s/-e/-E/g" $1
sed -i "s/-s/-S/g" $1
sed -i "s/-d/-D/g" $1
sed -i "s/ç/c/g" $1
sed -i "s/ô/o/g" $1
sed -i "s/ï/i/g" $1
sed -i "s/É/E/g" $1
sed -i "s/-p/-P/g" $1
sed -i "s/ë/e/g" $1
sed -i "s/î/i/g" $1
sed -i "s/û/u/g" $1
sed -i "s/ü/u/g" $1
sed -i "s/œ/oe/g" $1
sed -i "s/'h/'H/g" $1
sed -i "s/à/a/g" $1
sed -i "s/Î/I/g" $1
sed -i "s/-a/-A/g" $1
sed -i "s/Œ/Oe/g" $1
sed -i "s/ÿ/y/g" $1
sed -i "s/È/E/g" $1


Good thing is that it could also be used to fix accents in localadmins ;)

**Comment by stepps00** [comment]
Yes - the accents will probably present themselves in many of the `locality` and `localadmin` records. You may be able to use the `name:*_x_preferred` and  `name:*_x_variant` properties for variations of the `wof:name`.

**Comment by Joxit** [comment]
@adelcasse Yes it is, I updated the project, it's in nodejs so I used a proper package for this :stuck_out_tongue:

@stepps00 Yes I know, but [pelias-pip-service](https://github.com/pelias/pip-service) does not send all names variant :/

The archive has been updated, you can [download](https://media.r7.jawg.io/wof-utils/wof-fr-admin-express.zip) it again :wink:.

**Comment by stepps00** [comment]
@Joxit - thanks, will take a look!

**Comment by Joxit** [comment]
@stepps00 fine, hope this will be good now

With that, wof will be more up to date than osm

**Comment by adelcasse** [comment]
That's perfect, thanks a lot @Joxit !!!

**Comment by adelcasse** [comment]
@stepps00 any news on this or any help needed to process the changes ?

**Comment by stepps00** [comment]
@joxit - are you able to upload your most current file again? It looks like your most recent file (`https://media.r7.jawg.io/wof-utils/wof-fr-admin-express.zip`) is no longer available for download.

I have a local copy, but I want to make sure I'm working with the most recent changeset.

**Comment by stepps00** [comment]
I'm starting a PR today with updates to the `localadmin` and `locality` records in France. Here are notes about the update process (outlined above, but noting again for good measure):

- Verify data source
- Add source file, if necessary
- `newlocalities.geojson`:
  - process as completely new features. use import script for new features, map source properties and use min viable property set
- `localityandlocaladmin.geojson`:
  - load in locality file, update properties and geometries, save existing geometry as alt file, make coterminous with localadmin id.
  - load in localadmin file, update properties and geometries, save existing geometry as alt file, make coterminous with locality id.
- `localities.geojson`:
  - load in locality file, update properties and geometries, save existing geometry as alt file.
- `localadmin.geojson`:
  - load in localadmin file, update properties and geometries, save existing geometry as alt file.

Figure out if there are any localadmin/locality records that were not udpated or overlap new geometries.

**Comment by stepps00** [comment]
Also, @joxit and @adelcasse - I'm noticing features in the "newlocalities.geojson" file should map to existing WOF records, but are not. An example:

The feature with the `NOM_COM_M` value of "ESQUEHERIES" is suggested as a new record, but WOF already has it: https://spelunker.whosonfirst.org/id/101899821/. 

Can you explain how this existing record was not caught? I want to make sure we are not importing duplicate records at either placetype.

**Comment by stepps00** [comment]
@Joxit - thanks for the quick response! I'm going to take a look at the new geometries right now..

**Comment by Joxit** [comment]
Ok
I will try to check your PR tomorrow at 9 am CEST. 
Let me know if you need some help

**Comment by Joxit** [comment]
I'm trying to compute `newlocalities.geojson` in order to found duplicates (helped by pelias API).
I will provide a new link ASAP with new data.
I'll keep you informed.

**Comment by stepps00** [comment]
Hey @joxit - no rush on the data. I am still processing your other files and testing an import script locally.

**Comment by Joxit** [comment]
Hi @stepps00 
Sorry it took a little longer than expected but I updated the data.
There are fewer cities not found (those in `newlocalities.geojson`).
Now there are 508 new localities instead of 1108.
The link is [here](https://media.jawg.io/wof-utils/wof-fr-admin-express-2017.zip) with md5sum:
```
e3f5a7b8550689a534c8bf006490fe62  wof-fr-admin-express-2017.zip
```

**Comment by stepps00** [comment]
Thanks @joxit - will take a look!

**Comment by stepps00** [comment]
Hey @joxit - I am wrapping up the import of your files, via #1189. A few notes, as I'm verifying feature counts and gaps in coverage. Right now, Who's On First has 36,218 "current" localadmin records and 36,714 "current" locality records in France. 

### localadmin
Out of the 36,218 Who's On First localadmin records in France, 34,766 records matched with one of your features, but 1,452 records did not. 

I've verified quite a few of these non-matching records, like "Saint-Nicolas-Des-Laitiers", "Montfaucon-Montigne", and "Saint-Girod" and they are all referenced as "former communes". Do you expect these non-matching 1,452 records to be former communes? If so, I will add another commit to tag these 1,452 localadmin as non-current in Who's On First.


### locality
Out of the 36,714 Who's On First locality records in France, 4,454 records matched with one of your features, but 32,260 records did not. In addition, we have 508 "new" localities that were not already in Who's On First.

For this placetype, the count seems quite low. A few examples of locality records that did not match:

- [Vincy-Reuil-et-Magny](https://spelunker.whosonfirst.org/id/101898739)
- [Bourguignon-lès-Morey](https://spelunker.whosonfirst.org/id/101886419)
- [Champagne-sur-Vingeanne](https://spelunker.whosonfirst.org/id/101885305)
- [Saint-Martin-du-Mont](https://spelunker.whosonfirst.org/id/101888067)

Is this because the majority of locality records in France in Who's On First are actually localadmin? The above four examples are valid "communes" - do you expect France to have many more localadmin than locality records?

I think I understand the changes here, but I am curious about the update process before importing -- thanks again for all the help!

**Comment by adelcasse** [comment]
@stepps00 Do you have a list of non-matched localadmins ? As of January 1st 2018, France has 35.357 "communes" (but depending on how old was the data before the update, the decrease of localadmins records could be legitimate, see attached file
![nombre_communes](https://user-images.githubusercontent.com/403029/39663516-1875e7c6-5075-11e8-8a66-8f8da380bb25.png)
).

On localities, yes, I think it is expected that localities are actually localadmins (or both localities and localadmins). If you look at what guys from Qwant did recently using OSM data on Cosmogony here http://cosmogony.world/, there are very few "city districts" comparing to "cities" (which are the "communes"). And "city districts" are mostly (except in the main metropolitan areas) "former communes", in fact.

**Comment by Joxit** [comment]
Hi @stepps00. Sorry for the delay I was traveling.

### localadmin

Every year since 2011, cities disappear from France and "[Saint-Nicolas-Des-Laitiers](https://fr.wikipedia.org/wiki/Saint-Nicolas-des-Laitiers)", "[Montfaucon-Montigne](https://fr.wikipedia.org/wiki/Montfaucon-Montign%C3%A9)", and "[Saint-Girod](https://fr.wikipedia.org/wiki/Saint-Girod)" are some of them. On wikipedia they are tagged as `ancienne commune française`.
What France is currently doing is to merge several communes (`ancienne commune française` or "former communes") to form new ones (`nouvelle commune française`).
An example with "Saint-Nicolas-des-Laitiers", it was merged with 9 other municipalities ("Anceins", "Bocquencé", "Couvains", "La Ferté-Frênel", "Gauville", "Glos-la-Ferrière", "Heugon", "Monnai" and "Villers-en-Ouche" ) to form [La Ferté-Frênel](https://fr.wikipedia.org/wiki/La_Fert%C3%A9-en-Ouche).
That's why there are some missing localadmin.
However, the `ancienne commune française` often keep their boundaries, it might be possible to downgrade in locality but we don't have their correct shape...

### locality

Oh, there has been an issue here, I have not succeeded to match everything correctly. There is a precision issue in my requests.
In fact, I don't know the difference between locality and localadmin for Who's on First and don't know how this should be applied to France.
What I did was just a script to retrieve admin boundaries from shapefile data and see the WOF matches. And that seems to often miss for the localities... Sorry It was not my goal to remove localities :/

**Comment by Joxit** [comment]
Hi @adelcasse.
What did Qwant look cool. We could be inspired for WOF, but we may miss the data for the `ancienne communes` (I don't see them in admin express)

**Comment by adelcasse** [comment]
@Joxit as I understood it reading the placetypes définitions, localadmins are the administrative level of a city ("commune"), and localities are every boundaries that can be seen as cities. For example if you take the "commune" of Antibes, you will have Antibes (06600, both localadmin and locality, maybe with a smaller shape) and Juan-les-Pins (06160, as locality as it dépends from the commune Antibes). This might be the same with new/former communes and in big cities like Paris or Marseille that have "mairies d'arrondissement". In every other places (in most cases regarding the french administrative layers), locality and localadmin should be the same (and I don't know in that case if localities should be duplicated from localadmins or removed, @stepps00 ?)

**Comment by Joxit** [comment]
@adelcasse Ok, I see... But [Juan-les-Pins](https://fr.wikipedia.org/wiki/Juan-les-Pins) is tagged as "quartier" (neighbourhood) on wikipedia and [locality in WOF](https://spelunker.whosonfirst.org/id/1125921055/)
So, **localadmin** are all `commune française` (from [admin-express](http://professionnels.ign.fr/adminexpress)) for sure.
And **localities** should be all `ancienne commune française` and localities from big cities ?

@stepps00 I forgot something, in my export there is only metropolitan France (not the municipalities of the overseas regions like [Reunion](https://spelunker.whosonfirst.org/id/85671199/)). This is also why some localadmin are missing.

**Comment by adelcasse** [comment]
@Joxit For Juan-les-Pins (I live in Antibes that's why I took the example ;)), it's blur but concretely, the address usually written for a letter to someone in Juan-les-Pins is "XX rue de YYY, 06160 Juan-les-Pins" (or "XX rue de YYY, 06160 Antibes Juan-les-Pins") and not "XX rue de YY, 06600/06160 Antibes", By the way, even if the official name of the "localadmin" is Antibes, it is commonly named "Antibes Juan-les-Pins" (see the website of the city : http://www.antibes-juanlespins.com/). So I think it should (or at least could) be considered as a locality that's not a localadmin. 

Anyway, this was just to illustrate the thing with a place that is not a "former commune" ;)

**Comment by adelcasse** [comment]
The problem is that, if we don't fix localities, importing in Pelias using WOF will still not be accurate. On my first post and the example of Rougegoutte, that's the locality of Rougegoutte that is completely wrong (the localadmin is "better" but we match locality first). 

So maybe : 

- solution 1 : ignoring "former communes" for now in localities but still upgrading the existing localities ? But I don't know in such former communes how someone would search for an address
- find an older admin express (or the old name of this IGN data) and compare with the most recent one you used here (or a list of former communes) to insert former communes un localities

**Comment by stepps00** [comment]
Hey @adelcasse and @joxit - thanks for all the replies.. you can take a look at the proposed changes (so far), here: https://github.com/whosonfirst-data/whosonfirst-data/pull/1189

To address the new comments/questions:

- I will prepare a csv file with the "leftover" non-matching ids of localadmin/locality records in France.. expect that shortly.

- I have also not updated the overseas records for France (places like Reunion), so my counts do not include those records.

- You can take a look at the placetype definition for localities [here](https://github.com/whosonfirst/whosonfirst-placetypes#locality) and the definition for localadmin [here](https://github.com/whosonfirst/whosonfirst-placetypes#localadmin). Essentially. localities are towns and cities - places that parent neighbourhoods. Localadmin, on the other hand, often contain more than one locality.

I am happy to hear your input on whether or not these features fall into the "locality"  or "localadmin" placetype. It sounds to me like we may want to import all of these features as a single placetype, and, when applicable, using the `wof:placetype_alt` property to tag the additional property they represent.

**Comment by Joxit** [comment]
@adelcasse hum... You are right... That's not easy.

I think the second solution should be done by hand :disappointed: 

Hey @stepps00 sorry we flooded the issue :zipper_mouth_face: 

Ok, I will take a look tomorrow, it's getting late here :confused:  and thank you for your links.

If I understand correctly, we will have at the end only one WOF entry (one id) that will contain both types of place? This will significantly reduce WOF inputs and facilitate changes pour la suite.

I have updated (sorry again) the export [here](https://media.jawg.io/wof-utils/wof-fr-admin-express-2017.zip) with md5sum : 
```
0b747cec3f5c7960a11a64c7b5195190 wof-fr-admin-express-2017.zip
```
This new version fix the non matching localities like `Vincy-Reuil-et-Magny`. Now there are 30924 cities with their WOF locality + localadmin id.

**Comment by stepps00** [comment]
No problem at all, thanks @joxit !

**Comment by Joxit** [comment]
I looked at your PR #1200 a bit.
I saw that you use only localadmin. You intend to put the tag ` wof:placetype_alt` ?
The old localities will be deleted or tagged as unused/deprecated ?

What is your tool for GeoJSON formatting ?

**Comment by stepps00** [comment]
Hey @joxit - the plan of attack for this issue is going to be merging the `localadmin` placetype first. Then, once merged, validating the `wof:placetype_alt` property for the `localadmin` records and updating any localities as necessary.

Hoping to merge these new geometries soon -- stay tuned.

**Comment by nvkelso** [comment]
> What is your tool for GeoJSON formatting ?

@Joxit We use https://github.com/whosonfirst/py-mapzen-whosonfirst-export/blob/master/scripts/wof-exportify which makes working with GeoJSON diffs much easier.

**Comment by adelcasse** [comment]
@stepps00 I see that there is 1 remaining task in the PR #1200 : do you need any help on this ?

**Comment by stepps00** [comment]
Hey @adelcasse - thanks for your patience on this issue. This is on my to-do list - I'm expecting to have this merged by the end of the week :)

**Comment by stepps00** [comment]
@joxit and @adelcasse - We've merged #1257, thanks for the help with this one.

**Comment by Joxit** [comment]
Nice, thank you for your amazing work!
